build/includes/terraform.mk

```diff
@@ -16,54 +16,58 @@
-	cd $(mount_path)/install/terraform && terraform init && gcloud auth application-default login'
+	cd $(mount_path)/build/terraform/gke && terraform init && gcloud auth application-default login'
 
 terraform-clean:
-	rm -r ../install/terraform/.terraform
-	rm ../install/terraform/terraform.tfstate*
+	rm -r ../build/terraform/gke/.terraform || true
+	rm ../build/terraform/gke/terraform.tfstate* || true
 
 # Creates a cluster and install release version of Agones controller
 # Version could be specified by AGONES_VERSION
 gcloud-terraform-cluster: GCP_CLUSTER_NODEPOOL_INITIALNODECOUNT ?= 4
 gcloud-terraform-cluster: GCP_CLUSTER_NODEPOOL_MACHINETYPE ?= n1-standard-4
 gcloud-terraform-cluster: AGONES_VERSION ?= ''
+gcloud-terraform-cluster: GCP_CLUSTER_NAME ?= test-cluster
 gcloud-terraform-cluster: $(ensure-build-image)
 gcloud-terraform-cluster:
 ifndef GCP_PROJECT
 	$(eval GCP_PROJECT=$(shell sh -c "gcloud config get-value project 2> /dev/null"))
 endif
-	$(DOCKER_RUN) bash -c 'export TF_VAR_agones_version=$(AGONES_VERSION) && \
-		cd $(mount_path)/install/terraform && terraform apply -auto-approve -var values_file="" \
-		-var chart="agones" \
-	 	-var "cluster={name=\"$(GCP_CLUSTER_NAME)\", machineType=\"$(GCP_CLUSTER_NODEPOOL_MACHINETYPE)\", \
-		 zone=\"$(GCP_CLUSTER_ZONE)\", project=\"$(GCP_PROJECT)\", \
-		 initialNodeCount=\"$(GCP_CLUSTER_NODEPOOL_INITIALNODECOUNT)\"}"'
-	$(MAKE) gcloud-auth-cluster
+	$(DOCKER_RUN) bash -c 'cd $(mount_path)/build/terraform/gke && \
+		 terraform apply -auto-approve  -var agones_version="$(AGONES_VERSION)" \
+		-var name=$(GCP_CLUSTER_NAME) -var machine_type="$(GCP_CLUSTER_NODEPOOL_MACHINETYPE)" \
+		-var values_file="" \
+		-var zone="$(GCP_CLUSTER_ZONE)" -var project="$(GCP_PROJECT)" \
+		-var node_count=$(GCP_CLUSTER_NODEPOOL_INITIALNODECOUNT)'
+	GCP_CLUSTER_NAME=$(GCP_CLUSTER_NAME) $(MAKE) gcloud-auth-cluster
 
 # Creates a cluster and install current version of Agones controller
 # Set all necessary variables as `make install` does
+# Unifies previous `make gcloud-test-cluster` and `make install` targets
 gcloud-terraform-install: GCP_CLUSTER_NODEPOOL_INITIALNODECOUNT ?= 4
 gcloud-terraform-install: GCP_CLUSTER_NODEPOOL_MACHINETYPE ?= n1-standard-4
 gcloud-terraform-install: ALWAYS_PULL_SIDECAR := true
 gcloud-terraform-install: IMAGE_PULL_POLICY := "Always"
 gcloud-terraform-install: PING_SERVICE_TYPE := "LoadBalancer"
 gcloud-terraform-install: CRD_CLEANUP := true
+gcloud-terraform-install: GCP_CLUSTER_NAME ?= test-cluster
 gcloud-terraform-install:
 ifndef GCP_PROJECT
 	$(eval GCP_PROJECT=$(shell sh -c "gcloud config get-value project 2> /dev/null"))
 endif
 	$(DOCKER_RUN) bash -c ' \
-		cd $(mount_path)/install/terraform && terraform apply -auto-approve -var agones_version="$(VERSION)" -var image_registry="$(REGISTRY)" \
+	cd $(mount_path)/build/terraform/gke && terraform apply -auto-approve -var agones_version="$(VERSION)" -var image_registry="$(REGISTRY)" \
```

Comment:
Should this target install the latest compiled version, like [Make install](https://github.com/googleforgames/agones/blob/master/build/Makefile#L294) ?

Reply:
It got set using 
`agones_version="$(VERSION)"` above
```
GCP_CLUSTER_NAME=test-cl2 make gcloud-terraform-install
docker run --rm -v /Users/alekser/go/src/github.com/agones/agones/build//.config/gcloud:/root/.config/gcloud -v ~/.kube/:/root/.kube -v ~/.helm:/root/.helm -v /Users/alekser/go/src
/github.com/agones/agones:/go/src/agones.dev/agones -v /Users/alekser/go/src/github.com/agones/agones/build//.gomod:/go/pkg/mod -v /Users/alekser/go/src/github.com/agones/agones/bu
ild//.gocache:/root/.cache/go-build -e "KUBECONFIG=/root/.kube/config" -e "GO111MODULE=on" -w /go/src/agones.dev/agones  agones-build:8f473e7ff2 bash -c ' \                                             
        cd /go/src/agones.dev/agones/build/terraform/gke && terraform apply -auto-approve -var agones_version="1.5.0-39911ef" -var image_registry="gcr.io/agones-images" \                               
                -var pull_policy=""Always"" \
                -var always_pull_sidecar="true" 
```

Reply:
After that we are doing the same as in `make install`
https://github.com/googleforgames/agones/blob/5413b28aa7336603550c0de68d76461448f550f6/install/terraform/modules/helm/helm.tf#L91-L105
Setting helm parameter "agones.image.tag" to be an up to date `$(VERSION)`

Reply:
Tested this changes and the resulted controller pod uses correct image:
```
Controlled By:        ReplicaSet/agones-controller-55d9bdd8f7                                                                                                                                     [50/706]
Containers:
  agones-controller:
    Container ID:   docker://0a986a93103382805c304f837c68449719d47ac4390058effd1052e47719d466
    Image:          gcr.io/agones-images/agones-controller:1.5.0-39911ef
```
Registry could be changed by a `REGISTRY` make variable.

Reply:
Awesome - thanks for the explanation!